packages/cockpit/src/transactions/reprocess/cleanTransaction.js

```diff
@@ -33,15 +33,30 @@ const getCardId = ifElse(
 
 const getCustomerId = pathOr(null, ['customer', 'id'])
 
-const getCustomer = ifElse(
-  propEq('customer', null),
-  always(null),
-  pipe(
-    prop('customer'),
-    pick(['id'])
-  )
+const hasCustomerDocuments = hasPath(['customer', 'documents', 0])
+
+const buildCustomerObject = ({ address, phone, customer }) => ({
+  address,
+  phone,
+  ...customer
+})
+
+const customerBuilder = ifElse(
+  hasCustomerDocuments,
+  applySpec({
+    id: path(['customer', 'id'])
+  }),
+  buildCustomerObject
 )
 
+const getCustomer = (transaction) => {
+  if (transaction.customer === null) {
+    return null
+  }
+
+  return customerBuilder(transaction)
+}
```

Comment:
Dá para tocar um ramdinha aqui, ex:
```suggestion
const hasCustomer = pipe(
  prop('customer'),
  complement(isNil)
)

const g = ifElse(
  hasCustomer,
  customerBuilder,
  always(null)
)
```

Reply:
Fino, gostei.